packages/astro/src/@types/astro.ts

```diff
@@ -1685,3 +1732,7 @@ export type CreatePreviewServer = (
 export interface PreviewModule {
 	default: CreatePreviewServer;
 }
+
+export declare namespace AstroMiddleware {
```

Comment:
I tried this approach for the interface expansion inside a pet project, but unfortunately, that doesn't work :(

Any help would be appreciated!

Reply:
Fixed it!